File: db/models/team.py
import uuid
from datetime import datetime

from sqlalchemy import Column, DateTime, ForeignKey, String
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship

from db.database import Base


class Team(Base):
    __tablename__ = "teams"
    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    league_id = Column(UUID(as_uuid=True), ForeignKey("leagues.id"))
    name = Column(String, nullable=False)
    short_name = Column(String)
    logo_url = Column(String)
    created_at = Column(DateTime, default=datetime.utcnow)

    league = relationship("League", back_populates="teams")
    prediction_entries = relationship("PredictionEntry", back_populates="team")

File: tipssvenskan/lib/tipssvenskan/leagues/team.ex
defmodule Tipssvenskan.Leagues.Team do
  use Ecto.Schema
  import Ecto.Changeset

  schema "teams" do
    field :name, :string
    field :short_name, :string
    field :logo_url, :string

    belongs_to :league, Tipssvenskan.Leagues.League

    has_many :prediction_entries, Tipssvenskan.Predictions.PredictionEntry

    timestamps(type: :utc_datetime)
  end

  def changeset(team, attrs) do
    team
    |> cast(attrs, [:name, :short_name, :logo_url, :league_id])
    |> validate_required([:name, :league_id])
    |> unique_constraint([:league_id, :name])
    |> unique_constraint([:league_id, :short_name])
  end
end
